File: mobile/codeblurb/lib/data/content/models/coding_content_response.dart
import 'package:codeblurb/data/common/network/models/coding_content_type.dart';
import 'package:codeblurb/data/common/network/models/content_type.dart';
import 'package:codeblurb/data/content/models/test_case_response.dart';
import 'package:json_annotation/json_annotation.dart';

part 'coding_content_response.g.dart';

@JsonSerializable(createToJson: false)
class CodingContentResponse {
  final String name;
  final int id;
  final String description;
  final List<String> codeSkeleton;
  final List<String> codeSnippets;
  final List<TestCaseResponse> testCases;
  final CodingContentType codingContentType;

  CodingContentResponse(this.id, this.name, this.description, this.codeSkeleton,
      this.codeSnippets, this.testCases, this.codingContentType);

  ContentType get contentType => ContentType.coding;

  factory CodingContentResponse.fromJson(Map<String, dynamic> json) =>
      _$CodingContentResponseFromJson(json);
}

File: backend/codeblurb-backend/src/main/java/hu/codeblurb/backend/controller/dto/content/CodingContentResponse.java
package hu.codeblurb.backend.controller.dto.content;

import hu.codeblurb.backend.domain.content.CodingContent.CodingContentType;
import lombok.Getter;

import java.util.List;

import static hu.codeblurb.backend.domain.content.Content.ContentType.CODING;

@Getter
public class CodingContentResponse extends ContentResponse {

    private final Integer id;
    private final String description;
    private final List<String> codeSkeleton;
    private final List<String> codeSnippets;

    private final List<TestCaseResponse> testCases;

    private final CodingContentType codingContentType;

    public CodingContentResponse(Integer id, String name, String description, List<String> codeSkeleton, List<String> codeSnippets, List<TestCaseResponse> testCases, CodingContentType codingContentType) {
        super(name, CODING);
        this.id = id;
        this.description = description;
        this.codeSkeleton = codeSkeleton;
        this.codeSnippets = codeSnippets;
        this.testCases = testCases;
        this.codingContentType = codingContentType;
    }

    @Override
    public String toString() {
        return "CodingContentResponse[" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", description='" + description + '\'' +
                ", codeSkeleton=" + codeSkeleton +
                ", codeSnippets=" + codeSnippets +
                ", testCases=" + testCases +
                ']';
    }
}
